components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/LegacyJobOrchestrator.cs

```diff
@@ -63,13 +70,16 @@ public class LegacyJobOrchestrator : IJobOrchestrator {
         /// <param name="ct"></param>
         /// <returns></returns>
         public Task<JobProcessingInstructionModel> GetAvailableJobAsync(string workerId, JobRequestModel request, CancellationToken ct = default) {
-            _updated = false;
             if (_assignedJobs.TryGetValue(workerId, out var job)) {
                 return Task.FromResult(job);
             }
-            if ((job = _availableJobs.Dequeue()) != null) {
+            if (_availableJobs.Count() > 0 && (job = _availableJobs.Dequeue()) != null) {
```

Comment:
Use Count property instead of linq as it is faster (reads the count value, vs traversing).   More so though, here you could use TryDequeue which might even be faster as it checks and dequeues in one call.

Reply:
done